NumPy cheat sheet

hroncok opened the pull request:
First draft.

I have trouble using space effectively. I couldn't squeeze it all to A5, but on A4 it feels empty.

Feedback?

hroncok committed:
NumPy cheat sheet

hroncok commented:
Got some notes from @etellwen, will try to incorporate them.

hroncok committed:
NumPy: Adjust the pictures to be on one line

...instead of the previous chaos.

Based on feedback from @etellwen.

hroncok committed:
NumPy: Locations improvements

hroncok commented:
BTW I will need those on Wednesday morning.

encukou commented:
Tvorba polí Tvarem – do druhého sloupce bych dal např. `2×2 z nul`, aby bylo jasné co je ten tvar
Datové typy – druhý sloupec nesedí; asi bych tu nechal jen první a poslední, pro `U`/`a` napsal "řetězce znaků s omezenou délkou: `dtype=('U', 8)`"
Indexování  Na jedné dimenzi – místo "první prvek" bych dal "první řádek". (Pro jedno- nebo vícedimenzionální matice to není úplně nejlepší termín, ale na pochopení operace IMO ano.)
Indexování *n*-ticí – obrázky nesedí 

Nekonzistentní pojmenovávání – `a`, `array`, `matrix`, nic (`.astype()`) – nebude v tom guláš?

hroncok commented:
>Datové typy – druhý sloupec nesedí; 

aha, má tam být volná řádka, ale inkscape ji po uložení sežere

hroncok committed:
NumPy: 1st update after review

hroncok commented:
> Tvorba polí Tvarem – do druhého sloupce bych dal např. `2×2 z nul`, aby bylo jasné co je ten tvar

upraveno

> Datové typy – druhý sloupec nesedí; asi bych tu nechal jen první a poslední, pro `U`/`a` napsal "řetězce znaků s omezenou délkou: `dtype=('U', 8)`"

dal jsem místo toho všechny řádky, přijde mi to lepší,

> Indexování  Na jedné dimenzi – místo "první prvek" bych dal "první řádek". (Pro jedno- nebo vícedimenzionální matice to není úplně nejlepší termín, ale na pochopení operace IMO ano.)

napsal jsme první „řádek“

> Indexování *n*-ticí – obrázky nesedí 

které?

```pycon
>>> import numpy
>>> matrix = numpy.array([[1, 2], [3, 4]])
>>> matrix
array([[1, 2],
       [3, 4]])
>>> matrix[0, 1]
2
>>> matrix = numpy.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
>>> matrix
array([[1, 2, 3],
       [4, 5, 6],
       [7, 8, 9]])
>>> matrix[0:-1, 1:]
array([[2, 3],
       [5, 6]])
>>> matrix[:, 1]
array([2, 5, 8])
>>> cube = numpy.array([[[1, 2], [3, 4]], [[5, 6], [7, 8]]])
>>> cube
array([[[1, 2],
        [3, 4]],

       [[5, 6],
        [7, 8]]])
>>> cube[:, :, 0]
array([[1, 3],
       [5, 7]])
```

Napadá mě jen u té kostky, kdy nevím, jak se na to dívat, ale z hlediska té první kostky v cheatshetu je to stejně.

> Nekonzistentní pojmenovávání – `a`, `array`, `matrix`, nic (`.astype()`) – nebude v tom guláš?

Je to schválně. `.astype` jsem nahradil teda te za `a.astype`.

encukou commented:
> Indexování n-ticí – obrázky nesedí

Pardon, asi jsem se přehlédl :(

Nevím jestli se k tomu znovu dostanu včas; kdyžtak tiskni.

hroncok committed:
NumPy: Shift content on right/back page a bit up

Based on a real world print feedback

hroncok commented:
Alfa verze. Oboustranně a rozříznuto na A5.

![_20171030_150137](https://user-images.githubusercontent.com/2401856/32174917-7bc24930-bd83-11e7-819b-796e2e7266d1.JPG)

hroncok committed:
NumPy: Fix typo

encukou committed:
NumPy: Suggested edits

- Rearrange examples with pictures to keep text & picture together
- Get rid of empty() - we're leaving out more important things
  (IMO `indices()` would be more useful than `empty()`)
- Mention that there's another page (unlike other current cheatsheets)
- Move datatypes down
- add 'float64' example so it doesn't look like integers only
- Add OR example to combining truth tables (for non-C people)

encukou commented:
Kouknul jsem se na to a trochu to přeházel; můžeš to nechat nebo se jen inspirovat

hroncok committed:
NumPy: Further rearrangements

hroncok committed:
NumPy: Add @encukou to the footer

As per the edits not being minor

hroncok commented:
Rebased (commit messages only).

hroncok commented:
Udělal jsem posledních pár úprav.

Na straně dva jsem trochu čaroval s prázdným místem, podle mě to vypadá lépe.

Na straně jedna jsem nahradil velké `...` za text `a tak dále, počet dimenzí není omezený`

Přidal jsem @encukou do patičky.

Za mě dobrý.

hroncok commented:
Díky za feedback i úpravy.

encukou commented:
Díky za tahák! :)
